src/taipy/core/_scheduler/_scheduler.py

```diff
@@ -262,10 +194,19 @@ def __remove_jobs_to_run(cls, jobs):
 
     @classmethod
     def __cancel_jobs_and_processes(cls, job_id_to_cancel, jobs):
+        from ._dispatcher import _JobDispatcher
+
         for job in jobs:
-            if process := cls._dispatcher._pop_dispatched_process(job.id):
+            if process := _JobDispatcher._pop_dispatched_process(job.id):
                 process.cancel()  # TODO: this doesn't terminate the running process
             if job_id_to_cancel == job.id:
                 job.cancelled()
             else:
                 job.abandoned()
+
+    @staticmethod
+    def __check_and_execute_jobs_if_development_mode():
+        if Config.job_config.is_development:
+            from ._scheduler_factory import _SchedulerFactory
+
+            _SchedulerFactory._get_dispatcher()._execute_jobs_synchronously()
```

Comment:
Can we move this method to the development dispatcher directly?
My point is to not bother the Scheduler with this implementation constraint.

Reply:
we still need the scheduler to be the one instantiating the execution. I think we can move the method but we still need to import and use it in the scheduler which will involve duplicating the import code (the reason this function was written in the beginning)

Reply:
I agree that it is strange to have a method in scheduler only useful when it is a DevelopmentScheduler.